src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,442 @@
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="jit" />
+    
+    
+    
+    <!-- Get the crossgen and jit path-->
+    <PropertyGroup>
+      <_CoreclrPath>@(_CoreclrResolvedPath)</_CoreclrPath>
+      <_JitPath>@(_JitResolvedPath)</_JitPath>
+      <_CoreclrDir>$([System.IO.Path]::GetDirectoryName($(_CoreclrPath)))</_CoreclrDir>
+      <_CoreclrPkgDir>$([System.IO.Path]::Combine($(_CoreclrDir),"..\..\..\"))</_CoreclrPkgDir>
+      <_CrossgenDir>$([System.IO.Path]::Combine($(_CoreclrPkgDir),"tools"))</_CrossgenDir>
+      <!-- TODO override with rid specific tools path for x-arch -->
+      <_Crossgen>$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen"))</_Crossgen>
+      <_Crossgen Condition="$(OS.ToUpper().Contains('WINDOWS'))">$([System.IO.Path]::Combine($(_CrossgenDir),"crossgen.exe"))</_Crossgen>
+    </PropertyGroup>
+
+    <NETSdkError Condition="!Exists($(_Crossgen))"
+    ResourceName="UnableToFindResolvedPath"
+    FormatArguments="$(_Crossgen)" />
+
+    <PropertyGroup>
+      <_Crossgen> $([System.IO.Path]::GetFullPath($(_Crossgen)))</_Crossgen>
+    </PropertyGroup>
+
+    <!-- Create the output directory where crossgened assemblies are going to be present -->
+    <ItemGroup>
+      <_RuntimeOptimizedDirectories Include="@(_AssembliestoCrossGen -> '$(_RuntimeOptimizedDir)\%(RecursiveDir)')"/>
+    </ItemGroup>
+    <MakeDir  Directories="@(_RuntimeOptimizedDirectories)"/>
+
+    <!-- CrossGen the assemblies  -->
+    <Message Text="$(_Crossgen) -readytorun -in %(_AssembliestoCrossGen.Fullpath) -out $([System.IO.Path]::Combine($(_RuntimeOptimizedDir),%(RecursiveDir)%(Filename)%(Extension))) -jitpath $(_JitPath) -platform_assemblies_paths $(_RuntimeRefDir)$(PathSeparator)$(_NetCoreRefDir)"/>
+    <Exec
+          Command="$(_Crossgen) -readytorun -in %(_AssembliestoCrossGen.Fullpath) -out $([System.IO.Path]::Combine($(_RuntimeOptimizedDir),%(RecursiveDir)%(Filename)%(Extension))) -jitpath $(_JitPath) -platform_assemblies_paths $(_RuntimeRefDir)$(PathSeparator)$(_NetCoreRefDir)">
+      <Output TaskParameter="ExitCode" PropertyName="ErrorCode"/>
+    </Exec>
+
+
+    <!-- Copy by proxy the directory structure to PublishDir-->
+    <ItemGroup>
+      <_ManagedResolvedFileToPublish Include="$(_RuntimeOptimizedDir)\**\*.*" />
+    </ItemGroup>
+  </Target>
+
+
+  <!--
+    ============================================================
+                                        _GenerateCrossgenProj
+Generates the project used to restore netcoreapp correspoding to FX_Version
+    ============================================================
+    -->
+  <Target Name="_GenerateCrossgenProj">
+    <!-- Create a Project to Restore Crossgen -->
+    <PropertyGroup>
+      <_CrossgenProjContent>
+        <![CDATA[
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="$(MSBuildToolsVersion)">
+  <PropertyGroup>
+      <TargetFramework>$(_TFM)</TargetFramework>
+      <RuntimeIdentifier>$(RuntimeIdentifier)</RuntimeIdentifier>
```

Comment:
This is used to generate the Project to restore crossgen, the Alternative is to pass it as parameter to the Restore Target. I will leave it here for increased diagnosability